app/controllers/response_templates_controller.rb

```diff
@@ -0,0 +1,74 @@
+class ResponseTemplatesController < ApplicationController
+  after_action :verify_authorized, except: %i[index]
+
+  def index
+    not_found unless current_user
+    @response_templates = if params[:type_of] && params[:personal_included] != "true"
+                          result = ResponseTemplate.where(user_id: nil, type_of: params[:type_of])
+                          handle_authorization(result)
+                          result
+                        elsif params[:type_of] == "mod_comment" && params[:personal_included] == "true"
+                          result = ResponseTemplate.
+                            where(user_id: nil, type_of: "mod_comment").
+                            union(user_id: current_user.id, type_of: "personal_comment")
+                          handle_authorization(result)
+                          result
```

Comment:
Does the `authorize` in the handle_authorization [method](https://github.com/thepracticaldev/dev.to/pull/6823/files#diff-e6c099ad651ac2c674056e5f8340a483R71) raise an error if it can't be authorized?

This reads a little funny to me; is there any reason to not authorize the `result` and return it from the same method? (Rather than assigning it, passing it to the `handle_authorization` method, and then returning it from the `index` method?

Reply:
👍 all good thoughts here. Think I will refactor this after all 😓